diff --git a/tensorflow/BUILD b/tensorflow/BUILD
index 9d3fe6c6623..4fcb1f1fdbc 100644
--- a/tensorflow/BUILD
+++ b/tensorflow/BUILD
@@ -104,7 +104,8 @@ PACKAGE_STATIC_DEPS = [
 
 package(
     # copybara:uncomment default_applicable_licenses = [":license"],
-    default_visibility = [":internal"],
+    # default_visibility = [":internal"],
+    default_visibility = ["//visibility:public"],
 )
 
 # copybara:uncomment_begin
diff --git a/tensorflow/compiler/mlir/BUILD b/tensorflow/compiler/mlir/BUILD
index 42c78ff6143..30abee2eb75 100644
--- a/tensorflow/compiler/mlir/BUILD
+++ b/tensorflow/compiler/mlir/BUILD
@@ -102,10 +102,10 @@ cc_library(
 
 cc_library(
     name = "passes",
-    visibility = [
-        ":__subpackages__",
-        "//tensorflow/python:__subpackages__",
-    ],
+    # visibility = [
+    #     ":__subpackages__",
+    #     "//tensorflow/python:__subpackages__",
+    # ],
     deps = [
         "@llvm-project//mlir:AffineDialect",
         "@llvm-project//mlir:QuantOps",
